### PR TITLE
Remove Reader Improvements feature flag

### LINKED ISSUE #12026: Reader Improvements Phase 2 - Android Implementation Main Issue
This is a Main issue for the second phase of the Reader Improvements project on Android 
 pbArwn-fY-p2.

There are five main areas we'll need to cover:
- Introduce new "Pick a topic" screen - ([link](https://github.com/wordpress-mobile/WordPress-Android/issues/12027))
- Introduce new "Discover" tab with list of posts and a card with topic recommendations - ([link](https://github.com/wordpress-mobile/WordPress-Android/issues/12028))
- Introduce a new Repository layer between the newly written View+VM layers <-> the existing ReaderXyzActions + ReaderXyzServices + DB ([link](https://github.com/wordpress-mobile/WordPress-Android/issues/12039))
- ~Introduce new or~ Update the existing "Topic detail" screen - ([link](https://github.com/wordpress-mobile/WordPress-Android/issues/12318))
- Update the existing and add new endpoints ([discover endpoint](https://github.com/wordpress-mobile/WordPress-Android/issues/12319))

### PR DESCRIPTION
**Pull request by malinajirka**
Fixes #12026 

This PR removes the Reader Improvements Phase 2 feature flag -> it enables the feature on all builds.

Merge instructions
1. Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/12866 is merged
2. Remove "Not Ready for Merge" label

Please review each line I changed very carefully in this PR - it's quite easy to miss `!` or something similar and end up with a wrong condition. Thanks!

To test:
1. Smoke test Reader - make sure the new discover is shown

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,6 +1,8 @@
 15.7
 -----
- 
+ * [***] Reader: Introduced a new Discover tab tailored for each user.
```
Comment (malinajirka) [inline RELEASE-NOTES.txt:3]: Any suggestions for the release notes are very appreciated :).

File: WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java [hunk 2]
```diff
@@ -311,7 +311,7 @@ private void renderTagHeader(
                 onFollowButtonClicked,
                 ReaderTagTable.isFollowedTagName(currentTag.getTagSlug()),
                 isFollowButtonEnabled,
-                AppPrefs.isReaderImprovementsPhase2Enabled() || mAccountStore.hasAccessToken()
+                mAccountStore.hasAccessToken()
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java]: Tag header changes were released before RI FF flag was removed [here](https://github.com/wordpress-mobile/WordPress-Android/pull/12689). At that time we did not allow following tags locally for the logged out users. We can remove this access token check now.

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java]: Ohh nice catch! Thanks 🙇 

Fixed in dfd1c7101cfb52ffe782f0ec1f9aa32fe1f93f19

**Comment by ashiagr** [comment]
Thanks @malinajirka ! 
Changes look good to me 👍.

While testing I found a bug unrelated to these changes regarding follow/unfollow tags in a logged out scenario. Fixed it here: #12875.

**Review by ashiagr — APPROVED** [review]

**Pull request merged**